libs/cluster/Server/Migration/MigrationManager.cs

```diff
@@ -3,27 +3,57 @@
 
 using System.Collections.Generic;
 using System.Runtime.CompilerServices;
+using Garnet.common;
 using Garnet.server;
 using Microsoft.Extensions.Logging;
 
 namespace Garnet.cluster
 {
     internal sealed class MigrationManager
     {
+        readonly ILogger logger;
         readonly ClusterProvider clusterProvider;
         readonly MigrateSessionTaskStore migrationTaskStore;
 
+        /// <summary>
+        /// Used to initialize buffers for client connected to target node for active migrate sessions
+        /// </summary>
+        private NetworkBuffers networkBuffers;
+
+        /// <summary>
+        /// Get NetworkBuffers object
+        /// </summary>
+        public NetworkBuffers GetNetworkBuffers => networkBuffers;
+
         public MigrationManager(ClusterProvider clusterProvider, ILogger logger = null)
         {
-            migrationTaskStore = new MigrateSessionTaskStore(logger);
+            this.logger = logger;
+            this.migrationTaskStore = new MigrateSessionTaskStore(logger);
             this.clusterProvider = clusterProvider;
+            var bufferSize = 1 << clusterProvider.serverOptions.PageSizeBits();
+            this.networkBuffers = new NetworkBuffers(bufferSize, 1 << 12).Allocate(logger: logger);
         }
 
+        /// <summary>
+        /// Dispose
+        /// </summary>
         public void Dispose()
         {
-            migrationTaskStore.Dispose();
+            migrationTaskStore?.Dispose();
+            networkBuffers.Dispose();
         }
 
+        /// <summary>
+        /// Used to free up buffer pool
+        /// </summary>
+        public void Purge() => networkBuffers.Purge();
```

Comment:
It can be dangerous to call potentially mutating methods (purge sounds mutating, although its implementation here is safe as it probably passes through into the LFBP class). As this could be a potential copy of the struct, it can create subtle bugs, as well as correctness and maintenance challenges.

Reply:
Removed the wrapper and now we have a pool separate from NetworkBufferSettings